src/wp-includes/class-wp-classic-to-block-menu-converter.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * WP_Classic_To_Block_Menu_Converter class
+ *
+ * @package WordPress
+ * @since 6.3.0
+ */
+
+/**
+ * Converts a Classic Menu to Block Menu blocks.
+ *
+ * @since 6.3.0.
+ * @access public
+ */
+class WP_Classic_To_Block_Menu_Converter {
+
+	/**
+	 * Converts a Classic Menu to blocks.
+	 *
+	 * @since 6.3.0.
+	 * @param WP_Term $menu The Menu term object of the menu to convert.
+	 * @return string the serialized and normalized parsed blocks.
+	 */
+	public static function convert( $menu ) {
+
+		if ( ! is_nav_menu( $menu ) ) {
+			return new WP_Error(
+				'invalid_menu',
+				__( 'The menu provided is not a valid menu.' )
+			);
+		}
+
+		$menu_items = wp_get_nav_menu_items( $menu->term_id, array( 'update_post_term_cache' => false ) );
+
+		if ( empty( $menu_items ) ) {
+			return array();
+		}
+
+		// Set up the $menu_item variables.
+		// Adds the class property classes for the current context, if applicable.
+		_wp_menu_item_classes_by_context( $menu_items );
+
+		$menu_items_by_parent_id = static::group_by_parent_id( $menu_items );
+
+		$first_menu_item = isset( $menu_items_by_parent_id[0] )
+			? $menu_items_by_parent_id[0]
+			: array();
+
+		$inner_blocks = static::to_blocks(
+			$first_menu_item,
+			$menu_items_by_parent_id
+		);
+
+		return serialize_blocks( $inner_blocks );
+	}
+
+	/**
+	 * Returns an array of menu items grouped by the id of the parent menu item.
+	 *
+	 * @since 6.3.0.
+	 * @param array $menu_items An array of menu items.
```

Comment:
We generally separate `since` and `params` with an empty line (see [example](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-functions-class-methods))